databuilder/models/table_lineage.py

```diff
@@ -11,22 +11,15 @@
 from databuilder.models.table_metadata import ColumnMetadata, TableMetadata
 
 
-class TableLineage(GraphSerializable):
+class BaseLineage(GraphSerializable):
     """
-    Table Lineage Model. It won't create nodes but create upstream/downstream rels.
+    Generic Lineage Interface
     """
     LABEL = 'Lineage'
-    ORIGIN_DEPENDENCY_RELATION_TYPE = 'UPSTREAM'
-    DEPENDENCY_ORIGIN_RELATION_TYPE = 'DOWNSTREAM'
+    ORIGIN_DEPENDENCY_RELATION_TYPE = 'HAS_DOWNSTREAM'
```

Comment:
I think we should make it 

```
    ORIGIN_DEPENDENCY_RELATION_TYPE = 'DOWNSTREAM'
    DEPENDENCY_ORIGIN_RELATION_TYPE = 'UPSTREAM'
```

Will make it simple, and clear in terms of the same as UI. 

For Example:

origin: hive://gold.test_schema/test_table1
dependency: dynamo://gold.test_schema/test_table2

origin: hive://gold.test_schema/test_table1
dependency: hive://gold.test_schema/test_view1


hen the relation should look something like this.
```
origin -->[DOWNSTREAM] dependency
origin <--[UPSTREAM] dependency
```

Reply:
I agree, these semantics can be confusing 😕 

I do think it is important to have either a verb or preposition to help provide context to the words "upstream" and "downstream". I personally also agree with your directionality which is why I chose `HAS_DOWNSTREAM ` for the `origin --> dependency` relationship and `HAS_UPSTREAM` for the `dependency --> origin` relationship. By doing this, the nodes retrieved from a table using the `HAS_DOWNSTREAM` relationship will also show up in the downstream tab in the UI.

Another approach could be `origin -- [UPSTREAM_TO] --> dependency` and `dependency -- [DOWNSTREAM_OF] --> origin`. While this provides the same level of semantic context, retrieving a node with the `UPSTREAM_TO` relationship would actually show up in the downstream tab in the UI which could be slightly confusing.

Above all, I think that consistency here is probably the most important and would love to get input from a broader audience.

Reply:
When I read the relationship with the conventions proposed the semantic relationships don't seem to resonate with me, to me `DOWNSTREAM_TO` reads as if it is downstream, rather than the relationship created is downstream to the source. I put it into a quick picture just to baseline some of the options: 

![lineage](https://user-images.githubusercontent.com/10226663/115249388-8cd37780-a0f6-11eb-9175-d88e60b9c64b.png)

I'm happy to switch it up since consistency is really the key aspect here - I'll post this to the OSS chat to see if we can get some additional perspectives / votes.